app/src/main/java/org/p2p/wallet/auth/ui/done/AuthDonePresenter.kt

```diff
@@ -1,14 +1,24 @@
 package org.p2p.wallet.auth.ui.done
 
+import kotlinx.coroutines.launch
 import org.p2p.wallet.auth.interactor.UsernameInteractor
+import org.p2p.wallet.common.di.AppScope
 import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.push_notifications.ineractor.PushNotificationsInteractor
 
 class AuthDonePresenter(
-    private val usernameInteractor: UsernameInteractor
+    private val usernameInteractor: UsernameInteractor,
+    private val pushNotificationsInteractor: PushNotificationsInteractor,
+    private val appScope: AppScope
 ) : BasePresenter<AuthDoneContract.View>(), AuthDoneContract.Presenter {
 
     override fun load() {
         val username = usernameInteractor.getUsername()?.username
         view?.showUsername(username)
+
+        // Send device push token to NotificationService on creation and restoring the wallet
+        appScope.launch {
```

Comment:
I am not surely understand what's going on with our Scopes, we our AppScope contains Dispatchers.Main, even if it's so, our Presenter coroutine context also provide this dispatcher in default launch constructor. So i am little bit confused

Reply:
maybe we will not use custom scopes in presenter, and always will use default constructor, and inside suspend function wi will switch context of coroutine for example for ServiceScope